### PR TITLE
Eliminate fictitious "global" identifiers.

### PR DESCRIPTION
**Pull request by ptomin**
For instance, transform `global->ptr0001` to `g_ptr0001` or
`global->userVariable` to `userVariable`

### CODE REVIEW

**Comment by ptomin** [comment]
I have found 3 regression tests that never run during testing this feature:
https://github.com/uxmal/reko/tree/4603b2f6b49a6b4eb312530b645a05d1e47b716e/subjects/Elf/MIPS/animals
https://github.com/uxmal/reko/tree/4603b2f6b49a6b4eb312530b645a05d1e47b716e/subjects/PE/x86-64/load
https://github.com/uxmal/reko/tree/4603b2f6b49a6b4eb312530b645a05d1e47b716e/subjects/regressions/reko-15

File: src/Decompiler/Structure/FictitiousGlobalEliminator.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        public void Transform()
+        {
+            foreach (var stm in proc.Statements)
+            {
+                stm.Instruction = Transform(stm.Instruction);
+            }
+        }
+
+        public override Expression VisitFieldAccess(FieldAccess acc)
```
Comment (uxmal) [inline src/Decompiler/Structure/FictitiousGlobalEliminator.cs:52]: The logic is fine, but do you agree that it makes sense to add this logic to `ComplexExpressionBuilder` and `TypedConstantRewriter` instead? That would avoid injecting references to the temporary `global` in the first place.

Comment (ptomin) [reply src/Decompiler/Structure/FictitiousGlobalEliminator.cs]: Done in 8ae7402

**Comment by uxmal** [comment]
Nice initiative, Pavel. I may visit these classes and make some small changes but the logic is sound. The code looks much better -- and more correct.

**Pull request merged**